text/0000-no-bound-ergonomics.md

```diff
@@ -0,0 +1,99 @@
+- Feature Name: auto-bound
+- Start Date: 2018-06-24
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+This document adds a new ergonomics syntax to Rust to express type variables which *bounds* or
+*bounds combination* – are infered from the implementation – in order to *mirror* the already
+existing `impl Trait` in return position that infers the type on the implementation.
+
+```rust
+fn foo(x: &impl _) { // x’s bounds will be deduced from the implementation
```

Comment:
How is someone using this API supposed to know what they can pass to it?

Reply:
From the source code itself it is of course impossible. Outside of that, there are two principle ways that come to mind for knowing what they can pass in:

+ By reading the rustdoc, which can specify the actual bounds.
+ With IDE + RLS support which can tell you what the actual bounds are.

I'm reserving judgement on whether this is sufficient or a good idea for Rust in general.
I mostly use similar features in Haskell as a useful rapid prototyping development aid,
but I always specify the types after I've asked `ghci` what the types are.

Nonetheless, we don't have a proper REPL *yet*; so I don't know how valuable this would be until then.